From e1a7b4a6f3a5dde7865c81d89ec9b63f1b6406f6 Mon Sep 17 00:00:00 2001
From: Sakari Ailus <sakari.ailus@iki.fi>
Date: Sun, 28 Sep 2014 01:10:11 +0300
Subject: [PATCH 10/38] omap3isp: Add DT entry for OMAP 3

The resources the ISP needs are slightly different on 3[45]xx and 3[67]xx.
The range contains both.

Signed-off-by: Sakari Ailus <sakari.ailus@iki.fi>
---
 arch/arm/boot/dts/omap3.dtsi | 28 ++++++++++++++++++++++++++++
 1 file changed, 28 insertions(+)

diff --git a/arch/arm/boot/dts/omap3.dtsi b/arch/arm/boot/dts/omap3.dtsi
index d0e884d3a737..df38eb589f7f 100644
--- a/arch/arm/boot/dts/omap3.dtsi
+++ b/arch/arm/boot/dts/omap3.dtsi
@@ -197,6 +197,34 @@
 			};
 		};
 
+		omap3_isp: omap3_isp@480bc000 {
+			compatible = "ti,omap3-isp";
+			reg = <0x480bc000 0x0070 /* base */
+			       0x480bc400 0x01f0 /* ccp2 */
+			       0x480bc600 0x00a8 /* ccdc */
+			       0x480bca00 0x0048 /* hist */
+			       0x480bcc00 0x0060 /* h3a */
+			       0x480bce00 0x00a0 /* preview */
+			       0x480bd000 0x00ac /* resizer */
+			       0x480bd200 0x00fc /* sbl */
+			       0x480bd800 0x0170 /* csi2a (1st area) */
+			       0x480bd970 0x000c /* csiphy2 */
+			       0x480bd9c0 0x0040 /* csi2a (2nd area) */
+			       0x480bdc00 0x0170 /* csi2c (1st area) */
+			       0x480bdd70 0x000c /* csiphy1 */
+			       0x480bddc0 0x0040 /* csi2c (2nd area) */
+			       0x480022dc 0x0004 /* csirxfe */
+			       0x48002560 0x0004>; /* camera_phy_ctrl */
+			interrupts = <24>;
+			iommus = <&mmu_isp>;
+
+			#clock-cells = <1>;
+			ports {
+				#address-cells = <1>;
+				#size-cells = <0>;
+			};
+		};
+
 		gpio1: gpio@48310000 {
 			compatible = "ti,omap3-gpio";
 			reg = <0x48310000 0x200>;
-- 
2.1.4

